src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelSparkUtils.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param length the length of the index space
+     * @param numBlocks number of partitions (= blocks)
+     * @param minBlockSize minimum number of indices in each partition
+     * @return a list of {@link LinearSpaceBlock}
+     */
+    public static List<LinearSpaceBlock> createLinearSpaceBlocks(final int length, final int numBlocks,
+                                                                 final int minBlockSize) {
+        ParamUtils.isPositive(length, "The length of the linear space to be partitioned must be positive");
+        ParamUtils.isPositive(numBlocks, "The number of blocks must be positive");
+        ParamUtils.isPositive(minBlockSize, "Minimum block size must be positive");
+
+        final int blockSize = FastMath.max(length/numBlocks, minBlockSize);
+        final List<LinearSpaceBlock> blocks = new ArrayList<>();
+        for (int begIndex = 0; begIndex < length; begIndex += blockSize) {
+            blocks.add(new LinearSpaceBlock(begIndex, FastMath.min(begIndex + blockSize, length)));
+        }
+        /* the last block might be smaller than minBlockSize; we merge them */
+        while (blocks.size() > numBlocks && blocks.size() > 1) {
+            final int newBegIndex = blocks.get(blocks.size() - 2).getBegIndex();
+            final int newEndIndex = blocks.get(blocks.size() - 1).getEndIndex();
+            blocks.remove(blocks.size() - 1);
+            blocks.remove(blocks.size() - 1);
+            blocks.add(new LinearSpaceBlock(newBegIndex, newEndIndex));
+        }
+        return blocks;
+    }
+
+    /**
+     * Partitions an {@link INDArray} along the 0th axis according to a given partitioning and returns
+     * a list
+     *
+     * @param blocks partitioning
+     * @param arr an array
+     * @return list of partitioned arrays
+     */
+    public static List<Tuple2<LinearSpaceBlock, INDArray>> partitionINDArrayToAList(@Nonnull final List<LinearSpaceBlock> blocks,
+                                                                                    @Nonnull final INDArray arr) {
+        return blocks.stream().map(block ->
+                new Tuple2<>(block, arr.get(NDArrayIndex.interval(block.getBegIndex(), block.getEndIndex()))))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Partitions an {@link INDArray} along the 0th axis according to a given partitioning and returns
+     * a map
+     *
+     * @param blocks partitioning
+     * @param arr an array
+     * @return map of partitions to partitioned arrays
+     */
+    public static Map<LinearSpaceBlock, INDArray> partitionINDArrayToAMap(@Nonnull final List<LinearSpaceBlock> blocks,
+                                                                          @Nonnull final INDArray arr) {
+        return blocks.stream().map(block ->
+                new Tuple2<>(block, arr.get(NDArrayIndex.interval(block.getBegIndex(), block.getEndIndex()))))
```

Comment:
because spark is written in scala, and a `JavaPairRDD<K, V>` is basically a `JavaRDD<Tuple2<K, V>>` and creating `JavaPairRDD<K, V>`, as well as various map operations, require `Tuple2<K, V>` types.